src/scriptworker/cot/verify.py

```diff
@@ -1807,7 +1814,42 @@ def verify_repo_matches_url(repo, url):
     return True
 
 
-def get_source_url(obj):
+def valid_parent_source_urls(context, task):
+    """Return valid source urls.
+
+    Args:
+        context (scriptworker.context.Context): the scriptworker context.
+        task (dict): the task definition to check.
+
+    Raises:
+        CoTError: on error.
+
+    Returns:
+        tuple: valid source urls
+
+    """
+    source_env_prefix = context.config["source_env_prefix"]
+    repo_url = get_repo(task, source_env_prefix=source_env_prefix)
+    repo_url = repo_url.replace("git@github.com:", "ssh://github.com/", 1)
+    if repo_url.endswith(".git"):
+        repo_url = repo_url[:-4]
+    metadata_source_url = task["metadata"].get("source", "")
+    if is_github_task(task):
+        _, revision = extract_github_repo_and_revision_from_source_url(metadata_source_url, exception=CoTError)
```

Comment:
I don't think I like this idiom. If the caller wants a different exception, it should catch an re-raise, I think. That said, it isn't clear to me how much value the different exceptions types are providing us, we don't do appear to do anything with the exception type, and convert a bunch of random exception types to `CoTError` in `verify_chain_of_trust` anyway.